pkg/client/client.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package client provides a rest client to talk to the Trusty API.
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/stacklok/trusty-sdk-go/pkg/types"
+)
+
+const (
+	defaultEndpoint = "https://api.trustypkg.dev"
+	endpointEnvVar  = "MINDER_UNSTABLE_TRUSTY_ENDPOINT"
+	reportPath      = "v1/report"
+)
+
+// Options configures the Trusty API client
+type Options struct {
+	HttpClient netClient
+	BaseURL    string
+}
+
+// DefaultOptions is the default Trusty client options set
+var DefaultOptions = Options{
+	HttpClient: &http.Client{},
+	BaseURL:    defaultEndpoint,
+}
+
+type netClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// New returns a new Trusty REST client
+func New() *Trusty {
+	opts := DefaultOptions
+	if ep := os.Getenv(endpointEnvVar); ep != "" {
+		opts.BaseURL = ep
+	}
+	return NewWithOptions(opts)
+}
```

Comment:
This looks fairly minder specific. Should we keep this instantiation in Minder and make sure we call it with the appropriate options there?

Reply:
Sure, we can. I was trying to keep the logic here to make the env var work with any program but I didn't want to rename it to avoid changing it everywhere. 

But yeah, we can definitely instantiate it from minder through the options and I can use another name here.

Reply:
OK, I renamed the configuration envvar `TRUSTY_ENDPOINT` to be able to configure the client and, as discussed, I will preconfigure the client with its own var through the options when we move minder to use this one.